src/platform/win32bufferedfile.cpp

```diff
@@ -0,0 +1,290 @@
+/**
+ * @file
+ *
+ * @author OmniBlade
+ * @author xezon
+ *
+ * @brief Same as Win32LocalFile, but with buffered read and write. (Thyme Feature)
+ *
+ * @copyright Thyme is free software: you can redistribute it and/or
+ *            modify it under the terms of the GNU General Public License
+ *            as published by the Free Software Foundation, either version
+ *            2 of the License, or (at your option) any later version.
+ *            A full copy of the GNU General Public License can be found in
+ *            LICENSE
+ */
+#include "win32bufferedfile.h"
+#include <fcntl.h>
+
+namespace Thyme
+{
+
+Win32BufferedFile::Win32BufferedFile() : m_file(nullptr) {}
+
+Win32BufferedFile::~Win32BufferedFile()
+{
+    if (m_file != nullptr) {
+        fclose(m_file);
+        m_file = nullptr;
+        --s_totalOpen;
+    }
+
+    File::Close();
+}
+
+bool Win32BufferedFile::Open(const char *filename, int mode)
+{
+    int buffer_size;
+
+    if (!Decode_Buffered_File_Mode(mode, buffer_size)) {
+        return false;
+    }
+
+    if (!File::Open(filename, mode)) {
+        return false;
+    }
+
+    // r    open for reading (The file must exist)
+    // w    open for writing (creates file if it doesn't exist). Deletes content and overwrites the file.
+    // a    open for appending (creates file if it doesn't exist).
+    // r+   open for reading and writing (The file must exist).
+    // w+   open for reading and writing.
+    //      If file exists deletes content and overwrites the file, otherwise creates an empty new file.
+    // a+   open for reading and writing (append if file exists).
+
+    Utf8String modestr;
```

Comment:
The `fopen` function takes a string to control the file mode as opposed to flags. Below is my attempt to build this string based on the passed flags. I am not 100% sure I got it all right, but this is catered to the documentation above. It should be close enough. If something is off, we can tweak later or fix here in review.